### PR TITLE
feat: Delete User Route Added

### PR DESCRIPTION
**Pull request by manan04shah**
Fixed issue #24

### CODE REVIEW

File: routes/users/deleteUser.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"strconv"
+
+	"github.com/gofiber/fiber/v2"
+)
+
+// Delete a user based on the id as a query line parameter (empty body)
+
+func DeleteUser(c *fiber.Ctx) error {
+	//Raise an error if wrong request type
+	if c.Method() != "DELETE" {
```
Comment (RohanKhatua) [inline routes/users/deleteUser.go:16]: The route is registered as a delete route, all other http methods will automatically get rejected.

File: routes/users/deleteUser.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+	//Delete the user from the database
+	result := database.Database.Db.Delete(&models.User{}, userId)
+
+	if result.Error != nil {
+		log.Printf("Error deleting user: %v\n", result.Error)
+		return c.Status(500).SendString("Could not delete user - Internal Server Error")
+	} //If the user isn't deleted properly the function will return an error, and we'll know there was a problem during deletion
+
+	if result.RowsAffected == 0 {
```
Comment (RohanKhatua) [inline routes/users/deleteUser.go:36]: preferably, keep RowsAffected Check before err != nil, gorm quirks

**Review by RohanKhatua — CHANGES_REQUESTED** [review]
see review comms

**Comment by manan04shah** [comment]
changes have been made

**Review by RohanKhatua — APPROVED** [review]
LGTM

**Pull request merged**